### PR TITLE
remove check_tsc and refine cpu_dead

### PR DESCRIPTION
**Pull request by JasonChenCJ**
these two patches removed check_tsc function and refined cpu_dead function.
Tracked-On: #1842
Signed-off-by: Jason Chen CJ <jason.cj.chen@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
2068<br>No new violations to the coding guideline detected.

**Comment by HuiHuangShi** [comment]
2068<br>No new violations to the coding guideline detected.

File: hypervisor/arch/x86/timer.c [hunk 1]
```diff
@@ -286,3 +277,16 @@ void calibrate_tsc(void)
 	tsc_khz = (uint32_t)(tsc_hz / 1000UL);
 	printf("%s, tsc_khz=%lu\n", __func__, tsc_khz);
 }
+
+void check_tsc(void)
```
Comment (dongyaozu) [inline hypervisor/arch/x86/timer.c]: This patch is not necessary is general.

But the API name check_tsc is pretty confusing... It is kind of disable_rdtsc from user level.  HV doesn't have user level, so this code is a kind of "dead code" actually.

Either rename it, or removed it is fine to me.

**Comment by wenlingz** [comment]
@JasonChenCJ build fail please refer, also please add description on the conversation,thanks
************************************************************************
make[1]: Entering directory '/workspace/acrn-hypervisor/doc'

New errors/warnings found, please fix them:
==============================================

/workspace/acrn-hypervisor/doc/developer-guides/hld/hv-timer.rst:48: WARNING: doxygenfunction: Cannot find function "calibrate_tsc" in doxygen xml output for project "Project ACRN" from directory: doxygen/xml
make[1]: Leaving directory '/workspace/acrn-hypervisor/doc'
make[1]: *** [Makefile:57: html] Error 1
make: *** [Makefile:64: doc] Error 2

**Comment by HuiHuangShi** [comment]
2068<br>No new violations to the coding guideline detected.

**Pull request merged**